bitcoin/src/lib.rs

```diff
@@ -39,6 +39,7 @@
 // Exclude clippy lints we don't think are valuable
 #![allow(clippy::needless_question_mark)] // https://github.com/rust-bitcoin/rust-bitcoin/pull/2134
 #![allow(clippy::uninhabited_references)] // falsely claims that 100% safe code is UB
+#![allow(clippy::blocks_in_conditions)] // style lint, in bad taste
```

Comment:
Also an FP. The lint is supposed to detect something completely different.

Reply:
Oh, no, it's not FP. We actually are matching on a crazy complex expression. That truly is unreadable and we should assign it to a variable and match on that.

Reply:
I tried that and it was less readable.

Reply:
I tried it and I find it better. Is it possible that you did it differently? See how I did it in #2305

Reply:
Oops, I did try something different. I agree yours is better. Cherry-picked it into this PR.